src/site/content/en/blog/ta-types/index.md

```diff
@@ -0,0 +1,135 @@
+
+{% Img src="image/dPDCek3EhZgLQPGtEG3y0fTn4v82/1dxHGzPGPDgkMqRMj46o.jpeg", alt="A simplified depiction of integration testing showing how two unit working together.", width="800", height="450" %}
+
+{% Aside %}
+The tools you may consider using are often the same as those mentioned in the [unit testing section](#unit-testing) and can also include frameworks providing component testing, for example, [WebdriverIO](https://webdriver.io/) and [Cypress](https://www.cypress.io/).
+{% endAside %}
+
+### End-to-End testing
+
+These tests are often called UI tests, and this name explains their function even better. These tests interact with your application’s UI, including the complete application stack, and test your application from beginning to end.
+
+{% Img src="image/dPDCek3EhZgLQPGtEG3y0fTn4v82/9ZWWlop306SoGFP9rbeH.jpeg", alt="A simplified depiction of end-to-end testing showing a computer as a robot, looking at a workflow.", width="800", height="450" %}
+
+They resemble a system test if you refer to the theory of quality assurance. Thus, these tests simulate a genuine user and their interaction. These tests take more time because they involve the whole system. More runtime also means more computing power is needed and developers spending more time on waiting for pipelines to finish. Consequently, this additional effort results in higher maintenance costs. 
+
+{% Aside %}
+Possible tools to include (but not limited to) are:
+* [WebdriverIO](https://webdriver.io/) 
+* [Cypress](https://www.cypress.io/)
+* [Playwright](https://playwright.dev/)
+* [Selenium](https://www.selenium.dev/)
+{% endAside %}
+
+#### Visual UI testing
+
+An interesting subcategory of UI tests is visual tests. Those tests are extended end-to-end tests, providing a process of checking the visible output of an application by screenshot comparison: A screenshot of the change is compared with another screenshot containing the “status quo” - or in other words, a golden file -  and providing those results to a human reviewer to inspect and check. Put another way: it helps to find “visual bugs” in the appearance of a page, beyond purely functional bugs and outside the concept, not explicitly written down into assertions.
+
+### Static analysis
+
+There’s one more thing I want to introduce here: Static analysis is not a testing type in the textbook way. However, we’ll include it here as it will be an essential aspect in quality assurance  strategies later on. You can imagine it working like a spell check function: it scans your code for more significant defects and syntax errors without running the program, thus detecting code style issues. This simple measure can prevent many bugs. This is a good starting point to learn about Static Analysis if you want to get to know it in more detail.
```

Comment:
```suggestion
There’s one more thing we want to introduce here: static analysis is not a testing type in the textbook way. However, we’ll include it here as it will be an essential aspect in quality assurance  strategies later on. You can imagine it working like a spell check function: it scans your code for more significant defects and syntax errors without running the program, thus detecting code style issues. This simple measure can prevent many bugs. This is a good starting point to learn about Static Analysis if you want to get to know it in more detail.
```